synapse/api/ratelimiting.py

```diff
@@ -100,15 +100,17 @@ def can_do_action(
         if update:
             self.actions[key] = (action_count, time_start, rate_hz)
 
-        # Figure out the time when an action can be performed again
         if self.rate_hz > 0:
+            # Find out when the count of existing actions expires
             time_allowed = time_start + (action_count - burst_count + 1) / rate_hz
 
             # Don't give back a time in the past
             if time_allowed < time_now_s:
                 time_allowed = time_now_s
         else:
-            # This does not apply
+            # XXX: Why is this -1? This seems to only be used in
+            # self.ratelimit. I guess so that clients get a time in the past and don't
+            # feel afraid to try again immediately
```

Comment:
Note: I'm tempted to resolve this, but I don't want to change the behaviour of Ratelimiter in this PR anymore.

Reply:
I agree that this sounds like a follow-up!